src/hypertable.c

```diff
@@ -812,6 +812,16 @@ table_has_tuples(Oid table_relid, Snapshot snapshot, LOCKMODE lockmode)
 	return hastuples;
 }
 
+static bool
+table_is_logged(Oid table_relid, LOCKMODE lockmode)
+{
+	Relation	rel = heap_open(table_relid, lockmode);
+	bool		res = (rel->rd_rel->relpersistence == RELPERSISTENCE_PERMANENT);
```

Comment:
FWIW, there is a `RelationNeedsWAL(rel)` macro for this check in `utils/rel.h`

Reply:
I think the check as it is now is more clear. Leaving as-is

Reply:
Not sure I agree. Better to stick to available macros IMO, since that is probably what people intended for use and makes code easier to search (e.g., looking for all places where `RelationNeedsWAL` is called), as opposed to having everyone roll their own according to personal preference.